wandb/sdk/wandb_setup.py

```diff
@@ -322,6 +322,11 @@ def _setup(
 def setup(
     settings: Optional[Settings] = None,
 ) -> Optional["_WandbSetup"]:
+    """Set up a wandb session.
+
+    Note: This sets up a process-local singleton object.
+    (Forked processes will get a new copy of the object)
+    """
```

Comment:
I know this wasn't documented at all before---which is very bad for the public API---but this doc should be better. Off the top of my head, this docstring should answer:

* When should you use this?
* How do you use it correctly?
* How is the `settings` parameter used?
* What's the relationship to `wandb.init()`?
* What's the relationship to `wandb.teardown()`?

Reply:
done, please take a look!